### PR TITLE
Added Babili support

### PR DESCRIPTION
**Pull request by eagerestwolf**
This PR is to add support for the Babili minifer to Meteor, in accordance with meteor/meteor#8378. I will highlight the changes made since the diff will give the full details.

* Incremented minor version since this will be a major, but non-breaking change to the API as a whole.
* Added Babili 0.0 chain to packages.
* Added `getMinifier()` to `options.js` to get default minfier options.
* Minifer options are:
	- Do not generate a source map
	- Do not generate AST
	- Keep code in loose mode
* Added `meteorBabel.minify()` function.
* Added test for `meteorBabel.minify()` under `meteor babel`
* Updated `.travis.yml` to only test Node >=4.0, because Babili requires at least Node 4.0

### CODE REVIEW

File: options.js [hunk 1]
```diff
@@ -66,3 +66,20 @@ exports.getDefaults = function getDefaults(features) {
 
   return options;
 };
+
+exports.getMinifier = function getMinifier() {
```
Comment (abernix) [inline options.js]: I realize we're inside `options.js`, but my only ask might be a different name for this as it's not returning an actual minifier, but rather the default options.  Maybe `getMinifierDefaults`?

Comment (eagerestwolf) [reply options.js]: You actually are totally right, I started to do that and then I was like no that wouldn't be any good.

**Comment by eagerestwolf** [comment]
Updated and pushed.

**Comment by benjamn** [comment]
@sethmurphy18 Just wanted to let you know that I'll be on vacation for the next week (back Friday), and I'm sorry I didn't get to this before I left. I'll review/merge this PR if I get some downtime, but I'm not completely sure what the internet situation will be. Please direct questions to @abernix in the meantime! Thanks for working on this!

**Comment by eagerestwolf** [comment]
Hey no problem @benjamn, enjoy your vacation. @abernix and I have this under control.

**Review by abernix — APPROVED** [review]
These changes LGTM even before the minifier update (meteor/meteor#8378) gets completed.

Doesn't matter too much since it's server side, but slight increase to the dependency tree size thanks to `babel-preset-babili` (about 3%), but completely warranted given the removal of UglifyJS.  Overall it looks to be -5% since these are now combined.

I was a bit hesitant at first at the thought of stopping testing on Node `<4.0.0` since Meteor 1.3 still does depend on this package in its own `meteor-babel`, but seeing as Meteor 1.3 users are pinned to core releases of packages, this should be fine.  Meteor 1.3.x was the last release series on a pre-Node 4 version and we have no intentions of making additional releases on it.

Users of Meteor 1.3 will continue to use a version of this package which was tested at the time it was released.  It's hard to quantify if Meteor 1.3 users will  benefit from this anyhow since their Node version prohibits modern ECMAScript constructs anyhow.

**Comment by eagerestwolf** [comment]
I can fix that on the Meteor side of things, on the new version of the `meteor-babel` package, I can set the minimum API to 1.4, that way users of lower versions won't even be notified of the update.

**Comment by abernix** [comment]
> I can set the minimum API to 1.4

Not exactly sure what you're saying, but I'm not sure it matters.

All I was trying to say (in more of a conversation with myself, in possibly the wrong repo) is: For the purposes of this package getting published, we don't need to anything besides bump the [version in `babel-compiler`](https://github.com/meteor/meteor/blob/331e6e2c98fbad520a40e3f23039ed57c6d486be/packages/babel-compiler/package.js#L13) to match whatever this gets published as; We don't need to worry about Meteor 1.3.

**Comment by eagerestwolf** [comment]
I was referring to this:

```js
Package.onUse(function (api) {
  api.versionsFrom('1.4.0.0'); // <== this line here
  // other api stuffs
});
```

**Comment by abernix** [comment]
Ah, yeah, somewhat deceptively, that wouldn't accomplish preventing older versions from getting `babel-compiler` (if that's where you were going to put it).  That simply uses the default versions of _core_ packages which were bundled with the specified release of Meteor.  Also, we don't use `versionsFrom` in core packages.  

[The docs](http://docs.meteor.com/api/packagejs.html#PackageAPI-versionsFrom) explain a bit, though maybe not well enough.  The release itself dictates the dependencies which it will allow – although there have been varying degrees of success with different methods of implementing this.  Ultimately, Meteor 1.3 pins the versions very tightly to what was released at the time so they should be fine.  The [Meteor 1.4.3.1 blog post](https://blog.meteor.com/announcing-meteor-1-4-3-d1e9ab511d06) actually explains this somewhat helpfully in the "Stricter version constraints on core packages"  section (as we refined it again in Meteor 1.4.3).

**Comment by benjamn** [comment]
LGTM. Published [v0.16.0](https://github.com/meteor/babel/releases/tag/v0.16.0) to [npm](https://www.npmjs.com/package/meteor-babel).

**Pull request merged**